### PR TITLE
feat: View All Community and see details

### PR DESCRIPTION
**Pull request by GastonGutierrezC**
The view to see the communities and that users can subscribe to this, once subscribed can enter the community and begin to see its content.

Steps to test it: 

1. create one or more communities.
2. you must go to http://localhost:3000/communitiesUser , to see the user's communities.
3. you will be able to see the communities and the ones you have already subscribed to.
4. subscribe to a community
5. now if you go back to the page you will be able to see your community that you have just successfully subscribed to. 
6. enter a community you are already a part of and you will be able to see the messages that are in it.
7. now you can write messages and send images or gifs.

### CODE REVIEW

File: apps/frontend/app/communitiesUser/componets/LoadingMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const LoadingMessage = () => (
+    <p>Loading communities...</p>
+  );
+  
+  export default LoadingMessage;
+
```
Comment (Miguelito2774) [inline apps/frontend/app/communitiesUser/componets/LoadingMessage.tsx]: add end-line please.

File: apps/frontend/app/communitiesUser/componets/CommunityCard.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { Community } from "@/app/utils/Connections/connectionsCommunity";
+import React from "react";
+import Image from 'next/image';
+
+type Props = {
+  community: Community;
+  onClick: () => void;
+  buttonText?: string; // Propiedad opcional para el texto del botón
```
Comment (Miguelito2774) [inline apps/frontend/app/communitiesUser/componets/CommunityCard.tsx]: If there are going to be comments in the code, please make them in English and on a new line.

File: apps/frontend/app/communitiesUser/componets/CommunityList.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    <section className="w-full max-w-6xl mb-8">
+      <h2 className="text-xl font-bold mb-4">Your Communities</h2>
+      {communities.some((community) =>
+        // Verificar si el 'userId' está en la lista de 'community.users'
+        community.users.includes(userId) 
+      ) ? (
+        <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4">
+          {communities
+            .filter((community) =>
+              // Filtrar solo las comunidades donde el 'userId' está presente en 'community.users'
```
Comment (Miguelito2774) [inline apps/frontend/app/communitiesUser/componets/CommunityList.tsx]: If there are going to be comments in the code, please make them in English and on a new line.

File: apps/frontend/app/communitiesUser/componets/LoadingMessage.tsx (hunk 1 above)
Comment (Miguelito2774) [inline apps/frontend/app/communitiesUser/componets/LoadingMessage.tsx]: add end-line please.

File: apps/frontend/app/communitiesUser/yourCommunity/[id]/components/CommunityMessageBubble.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            }`}
+            style={{
+              clipPath: "polygon(50% 0%, 0% 100%, 100% 100%)",
+            }}
+          ></div>
+        </div>
+      </div>
+    );
+  };
+
```
Comment (Miguelito2774) [inline apps/frontend/app/communitiesUser/yourCommunity/[id]/components/CommunityMessageBubble.tsx:52]: add end-line please.

File: apps/frontend/app/utils/Connections/messageHistoryService.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import axios from "axios";
+
+const API_BASE_URL = "http://localhost:4444/messageHistory";
+
+export class MessageHistoryService {
+  // Obtener los mensajes de una comunidad
```
Comment (Miguelito2774) [inline apps/frontend/app/utils/Connections/messageHistoryService.ts]: If there are going to be comments in the code, please make them in English and on a new line.

File: apps/frontend/app/utils/Connections/messageHistoryService.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    try {
+      const response = await axios.get(`${API_BASE_URL}/community/${communityId}`);
+      return response.data;
+    } catch (error) {
+      console.error("Error fetching community messages:", error);
+      throw error;
+    }
+  }
+
+  // Enviar un nuevo mensaje
```
Comment (Miguelito2774) [inline apps/frontend/app/utils/Connections/messageHistoryService.ts]: Same here.

**Review by Miguelito2774 — CHANGES_REQUESTED** [review]

**Review by Miguelito2774 — CHANGES_REQUESTED** [review]
1. Add the number of members in the community.
2. Add or change a loading message instead of “You have no community”.
3. Enh: Have the admin or owner of the community have a special tag indicating that he/she is the owner.
4. Enh: Show the community owner on the community card.
5. Enh: Change “Community Messages” to the community name.

**Review by Miguelito2774 — APPROVED** [review]

**Review by Hairax — APPROVED** [review]

**Pull request merged**